src/de/gurkenlabs/litiengine/graphics/animation/AsepriteHandler.java

```diff
@@ -0,0 +1,412 @@
+    
+    JsonObject firstFrameObject = frames.getAsJsonObject().entrySet().iterator().next().getValue().getAsJsonObject();
+    JsonObject frameDimensions = firstFrameObject.get("sourceSize").getAsJsonObject();
+    
+    int frameWidth = frameDimensions.get("w").getAsInt();
+    int frameHeight = frameDimensions.get("h").getAsInt();
+    
+    return new Dimension(frameWidth, frameHeight);
+  }
+  
+  /**
+   * @param rootElement root element of JSON data.
+   * @param expected    expected dimensions of each key frame.
+   * @return true if key frames have same duration.
+   */
+  private static boolean areKeyFramesSameDimensions(JsonElement rootElement, Dimension expected) {
+    
+    JsonElement frames = rootElement.getAsJsonObject().get("frames");
+    
+    for (Map.Entry<String, JsonElement> entry : frames.getAsJsonObject().entrySet()) {
+      JsonObject frameObject = entry.getValue().getAsJsonObject();
+      JsonObject frameDimensions = frameObject.get("sourceSize").getAsJsonObject();
+      
+      int frameWidth = frameDimensions.get("w").getAsInt();
+      int frameHeight = frameDimensions.get("h").getAsInt();
+      
+      if (frameWidth != expected.getWidth() || frameHeight != expected.getHeight())
+        return false;
+    }
+    
+    return true;
+  }
+  
+  /**
+   * @param rootElement root element of JSON data.
+   * @return integer array representing duration of each key frame.
+   */
+  public static int[] getKeyFrameDurations(JsonElement rootElement) {
+    
+    JsonElement frames = rootElement.getAsJsonObject().get("frames");
+    
+    Set<Map.Entry<String, JsonElement>> keyFrameSet = frames.getAsJsonObject().entrySet();
+    
+    int[] keyFrameDurations = new int[keyFrameSet.size()];
+    
+    int frameIndex = 0;
+    for (Map.Entry<String, JsonElement> entry : keyFrameSet) {
+      JsonObject frameObject = entry.getValue().getAsJsonObject();
+      int frameDuration = frameObject.get("duration").getAsInt();
+      keyFrameDurations[frameIndex++] = frameDuration;
+    }
+    
+    return keyFrameDurations;
+  }
+  
+  /**
+   * Error that is thrown by the export class
+   */
+  public static class ExportAnimationException extends IOException {
+    public ExportAnimationException(String message) {
+      super(message);
+    }
+    
+    public ExportAnimationException(String message, Throwable cause) {
+      super(message, cause);
+    }
+    
+    public ExportAnimationException(Throwable cause) {
+      super(cause);
+    }
+  }
+  
+  /**
+   * Creates the json representation of an animation object and returns it.
+   * This is the public accesible function and can/should be changed to fit into the UI.
+   *
+   * @param spritesheetResource the animation object to export
+   * @throws ExportAnimationException if the export fails
+   */
+  public static String exportAnimation(SpritesheetResource spritesheetResource) throws ExportAnimationException {
```

Comment:
A sprite sheet can have multiple Animations. This method should be taking an animation.

Reply:
I agree! Don't confuse Animations (->keyframes) and Spritesheets (-> dimensions).

Reply:
I've tried to work on this PR myself  (https://github.com/Gamebuster19901/litiengine/tree/gson), but stopped as it requires changes to UtiLITI, as there is no way to select an Animation. You can only select a SpriteSheetResource.

As far as I know, `AssetPanelItem.origin` cannot be an `Animation`.

Reply:
IMO I think animations should be considered a resource themselves, because a spritesheet could have multiple animations, and an animation could be played on multiple spritesheets.

Reply:
> I've tried to work on this PR myself (https://github.com/Gamebuster19901/litiengine/tree/gson), but stopped as it requires changes to UtiLITI, as there is no way to select an Animation. You can only select a SpriteSheetResource.
> 
> As far as I know, `AssetPanelItem.origin` cannot be an `Animation`.

I think I'll make another attempt with this PR soon, since UtiLITI is in this repository now, and that makes development a lot easier.